packages/store-sync/src/recs/syncToRecs.ts

```diff
@@ -59,9 +64,46 @@ export async function syncToRecs<TConfig extends StoreConfig = StoreConfig>({
 
   world.registerDisposer(stopSync);
 
+  const getResourceSelector = async (functionSelector: Hex): Promise<Hex> => {
+    const entity = encodeEntity<
+      KeySchema & {
+        functionSelector: "bytes4";
+      }
+    >(components.FunctionSelectors.metadata.keySchema, {
+      functionSelector,
+    });
+
+    const selectors = getComponentValue(components.FunctionSelectors, entity);
+
+    // If we can't find selectors due to not being synced yet, we can try to read them from the world contract
+    if (!selectors) {
+      // TODO make fieldLayout a table metadata field
+      const encodedFieldLayout = valueSchemaToFieldLayoutHex(components.FunctionSelectors.metadata.valueSchema);
+
+      const [selectorRecord, encodedLengths, dynamicData] = await publicClient.readContract({
+        address: address as Hex,
+        abi: IBaseWorldAbi,
+        functionName: "getRecord",
+        args: [components.FunctionSelectors.id as Hex, [entity as Hex], encodedFieldLayout],
+      });
+
+      const decodedSelectors = decodeValue<
+        ValueSchema & {
+          resourceSelector: "bytes32";
+          systemFunctionSelector: "bytes4";
+        }
```

Comment:
same as above, this should be inferred from `components.FunctionSelectors.metadata.valueSchema`